idx/metric_definition.go

```diff
@@ -0,0 +1,552 @@
+package idx
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/binary"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/raintank/schema"
+	log "github.com/sirupsen/logrus"
+	"github.com/tinylib/msgp/msgp"
+)
+
+//go:generate msgp
+
+func init() {
+	msgp.RegisterExtension(95, func() msgp.Extension { return &MetricName{} })
+	msgp.RegisterExtension(90, func() msgp.Extension { return &TagKeyValues{} })
+}
+
+type mType uint8
+
+// MType represented by a uint8
+const (
+	MTypeUndefined mType = iota
+	MTypeGauge
+	MTypeRate
+	MTypeCount
+	MTypeCounter
```

Comment:
are count and counter different? maybe a comment would make sense to explain the difference, because that's not obvious

Reply:
I don't know actually. @Dieterbe any difference?